docs/5.0/application-access.md

```diff
@@ -0,0 +1,264 @@
+  auth_token: 4c7e15
+  auth_servers:
+  - 127.0.0.1:3025
+auth_service:
+  enabled: "yes"
+  listen_addr: 0.0.0.0:3025
+  tokens:
+  - proxy,node,app:4c7e15
+ssh_service:
+  enabled: "false"
+proxy_service:
+  enabled: "yes"
+  listen_addr: 0.0.0.0:3023
+  web_listen_addr: 0.0.0.0:3080
+  tunnel_listen_addr: 0.0.0.0:3024
+  public_addr: teleport.asteroid.earth:3080
+  ## Example using a wildcard cert.
+  https_keypairs:
+  https_keypairs:
+  - key_file: /etc/letsencrypt/live/teleport.example.com/privkey.pem
+  - cert_file: /etc/letsencrypt/live/teleport.example.com/fullchain.pem
+  - key_file: /etc/letsencrypt/live/*.teleport.example.com/privkey.pem
+  - cert_file: /etc/letsencrypt/live/*.teleport.example.com/fullchain.pem
+```
+#### [ Optional ] Obtain a new App Token
+In the above example we've hard coded a join token 4c7e15. You can use this to join apps or create a dynamic token using the tctl command below.
+```bash
+$ tctl tokens add --type=app
+```
+
+| Variable to replace | Description  |
+|-|-|
+| `auth_servers` | Address of Auth or Proxy Service setup above |
+| `auth_token` | Token used to connect to other Teleport processes  |
+
+```yaml
+teleport:
+  # nodename of machine running process.
+  nodename: wellington
+  # The Auth token `tctl tokens add --type=app` or the static token. 4c7e15
+  auth_token: "4c7e15"
+  # This is the location of the Teleport Auth Server or Public Proxy
+  auth_servers:
+    - teleport.example.com:3080
+auth_service:
+  enabled: no
+proxy_service:
+  enabled: no
+ssh_service:
+   enabled: no
+# The app_service is new
+app_service:
+   enabled: yes
+    debug_app: true
+   apps:
+   - name: "internal-app"
+     uri: "http://10.0.1.27:8000"
+   - name: "kubernetes-dashboard"
+     # This version requires a public_addr for all Apps, these
+     #  applications should have a certificate and DNS setup
+     public_addr: "example.com"
+     # Optional Labels
+     labels:
+        name: "jwt"
+     # Optional Dynamic Labels
+     commands:
+     - name: "os"
+       command: ["/usr/bin/uname"]
+       period: "5s"
+   - name: "arris"
+     uri: "http://localhost:3001"
+     public_addr: "arris.example.com"
+   # Teleport Application Access can be used to proxy any HTTP Endpoint
+   # Note: Name can't include any spaces
+   - name: "hackernews"
+     uri: "https://news.ycombinator.com"
+     public_addr: "hn.example.com"
+```
+
+## Advanced Options
+
+### Customize Public Address
+
+```yaml
+   - name: "jira"
+     uri: "https://localhost:8001"
+     public_addr: "jira.example.com"
+```
+
+### Deeplink to Subdirectory
+Some Applications are available on a Subdirectory, examples include Kubernetes Dashboard.
+
+```yaml
+   - name: "k8s"
+     uri: "http://10.0.1.60:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/#/overview?namespace=default"
+     public_addr: "k8s.example.com"
+```
+### Rewrite
+We provide simple rewrites. This is helpful for applications
```

Comment:
add more text, what applications, for example `This is helpful for applications that redirect to a different hostname. You can override the redirect to point back to teleport public address`